pkg/skaffold/hooks/container.go

```diff
@@ -83,17 +86,29 @@ func (h containerHook) run(ctx context.Context, out io.Writer) error {
 				args := []string{p.Name, "--namespace", p.Namespace, "-c", c.Name, "--"}
 				args = append(args, h.cfg.Command...)
 				cmd := h.cli.Command(ctx, "exec", args...)
-				cmd.Stderr = out
-				cmd.Stdout = out
+				tr, tw := io.Pipe()
+				cmd.Stderr = tw
+				cmd.Stdout = tw
 				podName := p.Name
 				containerName := c.Name
 				errs.Go(func() error {
+					defer tw.Close()
 					err := util.RunCmd(cmd)
 					if err != nil {
 						return fmt.Errorf("hook execution failed for pod %q container %q: %w", podName, containerName, err)
 					}
 					return nil
 				})
+				pod := p
+				var containerStatus v1.ContainerStatus
+				for _, status := range pod.Status.ContainerStatuses {
+					if status.Name == c.Name {
+						containerStatus = status
+					}
+				}
+				errs.Go(func() error {
+					return stream.StreamRequest(ctx, out, h.formatter(pod, containerStatus, func() bool { return false }), tr)
```

Comment:
I deliberately don't use the application logs `mute` settings, since the container hooks executes in sync phase and is totally unrelated to deploy settings other than the common podname prefix. WDYT?

Reply:
hmm, I see where you're coming from but I'm not sure we want to bypass these settings here. my reasoning is that if someone intentionally mutes the deploy logs via `--mute-logs=deploy`, they would expect hooks output to be muted as well - and this is the only way this output would get muted. so I'd be inclined to share the same `mute` setting.

alternatively, we could expose a separate `mute` setting for hooks, though this seems a bit overkill and I can't really imagine a scenario where a user would mute deploy logs but not sync hook logs.

Reply:
we definitely can't use the deploy `mute` setting since the application logs are manually muted in-between dev loops using this setting. This will have the effect of never actually printing the container hook log messages, since the container hooks are always executed in-between dev loops. IMO we can revisit a different setting for turning off the `hooks` output altogether.

Reply:
ahh right that makes sense. ok I filed https://github.com/GoogleContainerTools/skaffold/issues/6344 to track

Reply:
thanks!